### ISSUE TITLE
Only visited 1.18 chunks are rendered after Spigots --forceUpgrade

### ISSUE THREAD

**Issue by gobo7793** [overviewer/Minecraft-Overviewer#2002]
On my Server we recently updated Spigot to 1.18. I also did as mentioned on the release notes post on Spigots Homepage a server start using the `--forceUpdgrade` start parameter to update the whole worlds to the new chunk format. After that I pulled the current Overviewer master branch after merged PR #1990.

Since then, only chunks are rendered that were visited by a player since the update to 1.18. It doesn't matter if I start the rendering without options, `--check-tiles` or a fullrender of the world. The render of the world also doesn't show any errors for the world I want to render. It also shows the render like before, no blocks under y=0 (and we confirmed already that there should be blocks).

Example render:
![grafik](https://user-images.githubusercontent.com/9772185/144923122-0c700f2d-ace7-4751-84f7-5fa23f759984.png)

Overviewer Version: 0.3.1
Commit-ID I tried the last days: a5b36f5

Has anyone an idea what to do (except for visiting the whole world manually again)?

**Comment by CounterPillow** [comment]
Can you upload a zipped up region file here that contains some chunks with the issue here? I suspect I may know what the problem is (which chunk statuses we cull from rendering), but I need to look into the nbt to know how to fix it.

**Comment by gobo7793** [comment]
Of course, that's one of the region files rendered on the screeshot: [r.-2.-4.mca.zip](https://github.com/overviewer/Minecraft-Overviewer/files/7669507/r.-2.-4.mca.zip)

**Comment by urda** [comment]
I can confirm that we're also only seeing visited chunks on ours. I had started a re-render on our server to see if that would have corrected it. A user had noted this "section" had gone missing:

![image](https://user-images.githubusercontent.com/1648492/145152951-7debd3c6-8219-4be1-9f41-54365d119378.png)

Once I had a player travel between the two points in `1.18`, the render picked them up normally:

![image](https://user-images.githubusercontent.com/1648492/145153059-cf5b7579-1f61-4a17-b5ab-1540581b1e50.png)

But that area around it, used to be filled in prior to `1.18`. I can see about getting copies of chunks if you need additional sources.

**Comment by dalandis** [comment]
The launch of the chunky(plugin) helped us. After that, the map starts to display as it should.

**Comment by kqfflo** [comment]
I am also having the same issue. Chunky would not help I think. As I do not need to genereate squares or triangles. Any solutions / workaround yet?

**Comment by gobo7793** [comment]
The best workaround to generate the map is currently just to visit the chunks ingame

**Comment by Nekzuris** [comment]
> The best workaround to generate the map is currently just to visit the chunks ingame

That's rather unpractical with big maps, I also encounter this issue and I'm sure it's fixable since other block by block renderer like unmined or Chunky work just fine with unvisited chunk

**Comment by dalandis** [comment]
> I am also having the same issue but with Dynmap. Chunky would not help I think. As I do not need to generate squares or triangles. Any solutions / workarounds yet?

chunky works great. http://map.gmgame.ru/
there is a clear border of the transition from the old generation to the new

**Comment by gobo7793** [comment]
What exactly did you done? Just starting to fill the world until your border?

**Comment by dalandis** [comment]
I have a vanilla border, i did like this:
/chunky world nameWorld
/chunky worldborder
/chunky start

**Comment by gobo7793** [comment]
Okay, I tried it in the last days for our world and it looks that this would work as a workaround. But chunky only process around 2 or 3 chunks per seconds, which is (in our case) for a world with 100 km²... slow...

**Comment by greatmastermario** [comment]
I'm going to split my reply into two parts, one solved, one not.

Solved: Blocks below Y=0 and above Y=256 are generating with this PR: https://github.com/overviewer/Minecraft-Overviewer/pull/2012. If you try rendering with this change, the chunks loaded after 1.18 will have all blocks show

Not solved: Missing chunks. This is an issue caused by 1.18 updating the chunks to the new layout only after loading them in game. Once you get to certain distance from a 1.17 chunk, it will load as a proto-chunk, resetting its Status tag to "structure_starts". This results in Overviewer skipping this chunk since it thinks the chunk is not real. Normally this would be the case if the proto-chunk is on the edge of the world, but since this protochunk has existing data and is in the middle of the world, it results in the effects you see above. I do not know the proper way to handle this, but I can open a PR if we decide on a solution. See world.py line 1733-1735

**Comment by urda** [comment]
Are there any updates on supporting / correcting this ?

**Comment by Phoenix616** [comment]
> This results in Overviewer skipping this chunk since it thinks the chunk is not real. Normally this would be the case if the proto-chunk is on the edge of the world, but since this protochunk has existing data and is in the middle of the world, it results in the effects you see above. I do not know the proper way to handle this

Well checking the chunk for data or [biomes](https://github.com/overviewer/Minecraft-Overviewer/blob/master/overviewer_core/world.py#L1738) instead of using the lag could be a solution?
